### PR TITLE
Add action to run black on main

### PR DESCRIPTION
**Pull request by curufinwe**
Copy & paste from i6_core

### CODE REVIEW

**Comment by albertz** [comment]
You should also copy the pyproject.toml.

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+        cache: 'pip'
+    - run: pip install black==22.3.0
+    - run: black --diff .
+    - run: black --check .
```
Comment (michelwi) [inline .github/workflows/black.yml:20]: Actually 
```suggestion
    - run: black --diff --check .
```
works as well.
plus I guess we want a newline at the end of the file?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[tool.black]
+line-length = 120
+target-version = ["py37"]
+exclude = 'files'
```
Comment (albertz) [inline pyproject.toml:4]:
```suggestion
exclude = 'files'

```

File: .github/workflows/black.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  check-black-formatting:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+        cache: 'pip'
+    - run: pip install black==22.3.0
+    - run: black --diff --check .
```
Comment (albertz) [inline .github/workflows/black.yml:19]:
```suggestion
    - run: black --diff --check .

```

**Review by michelwi — APPROVED** [review]

**Review by Atticus1806 — APPROVED** [review]

**Pull request merged**